talk/morelanguage/morestl.tex

```diff
@@ -424,3 +424,59 @@
     \end{cppcode*}
   \end{exampleblock}
 \end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{\cpp date and time utilities \hfill \cpp11 / \cpp20}
+  \cppinline{std::chrono library} present in \cppinline{<chrono>} header
+  \begin{block}{Clocks}
+    \begin{itemize}
+    \item consists of a starting point (or epoch) and a tick rate
+      \begin{itemize}
+      \item E.g. January 1, 1970 and every second
+      \end{itemize}
+    \item C++ defines several clock type
+      \begin{itemize}
+      \item \href{https://en.cppreference.com/w/cpp/chrono/system_clock}{\color{blue!50!white} \cppinline{system_clock}} system time, aka wall clock time, or C time
+      \item \href{https://en.cppreference.com/w/cpp/chrono/steady_clock}{\color{blue!50!white} \cppinline{steady_clock}} monotonic but unrelated to wall clock time
+      \item \href{https://en.cppreference.com/w/cpp/chrono/high_resolution_clock}{\color{blue!50!white} \cppinline{high_resolution_clock}} clock with the smallest tick period
+      \end{itemize}
+    \end{itemize}
+  \end{block}
+  \begin{block}{\href{https://en.cppreference.com/w/cpp/chrono/time_point}{\color{blue!50!white} \cppinline{time_point}} and \href{https://en.cppreference.com/w/cpp/chrono/duration}{\color{blue!50!white} \cppinline{duration}}}
+    \begin{itemize}
+    \item provide easy manipulation of times and duration
+    \item clock dependent
+    \item \href{https://en.cppreference.com/w/cpp/chrono/duration/duration_cast}{\color{blue!50!white} \cppinline{duration_cast}} allows conversions between duration types
+      \begin{itemize}
+      \item available helper types : nanoseconds, microseconds, milliseconds, seconds, minutes, hours, ...
+      \end{itemize}
+    \end{itemize}
+  \end{block}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[11]{Practical usage / timing some \cpp code}
+  \begin{exampleblockGB}{How to measure the time spent in some code}{https://godbolt.org/z/PzKWer5eb}{\texttt{timing}}
+    \small
+    \begin{cppcode*}{gobble=2}
+      #include <chrono>
+
+      std::chrono::high_resolution_clock clock;
+      clock::time_point start = clock::now();
+      ... // code to be timed
+      std::chrono::duration<float> ticks = clock::now() - start;
+
+      auto millis = std::chrono::duration_cast
+                    <std::chrono::milliseconds>(duration);
+      std::cout << "it took " << duration.count() << " ticks"
+                << ", that is " << millis.count() << " ms\n";
+    \end{cppcode*}
```

Comment:
As an alternative, on could use `operator/(std::chrono::duration, ...)`. What do you think? The godbolt link shows an example of that.

```suggestion
  \begin{exampleblockGB}{How to measure the time spent in some code}{https://godbolt.org/z/avqj4hnMK}{\texttt{timing}}
    \small
    \begin{cppcode*}{gobble=2}
      #include <chrono>
      using namespace std::chrono_literals;

      std::chrono::high_resolution_clock clock;
      auto start = clock.now();
      std::this_thread::sleep_for(std::chrono::milliseconds(345));
      std::chrono::duration<float> ticks = clock.now() - start;

      std::cout << "it took " << ticks.count() << " ticks"
                << ", that is " << ticks / 1ms << " ms\n";
    \end{cppcode*}
```

Reply:
I hesitated, I did not have it initially, then added, then realized we did not present literals syntax and literal operators so it would look weird (did to me the first time), so I removed it again.

It's always the question of whether you present just recipes aka "black magic you should trust", or whether you build knowledge from scratch and explain everything you use before using it. I do not know

Btw : I've realized I forgot to push last version (with the fixes you had to do to compile). I've now pushed it.

Reply:
OK, ready to merge! 👍

Now we can think about whether we want to present literals somewhere ... 🤷‍♂️ 🙂

Reply:
:+1:  and #525 so that we do not forget